numpy/core/src/npymath/npy_math_internal.h.src

```diff
@@ -681,28 +632,33 @@ NPY_INPLACE @type@
 npy_remainder@c@(@type@ a, @type@ b)
 {
     @type@ mod;
-    if (NPY_UNLIKELY(!b)) {
-        mod = npy_fmod@c@(a, b);
-    } else {
+    if (!b) {
+        /*
+         * b == 0 or is NaN. `fmod` will return the right thing (NaN)
+         * but set the error flags correct (divmod may set too many).
+         * (Using !b here, since reverse logic set incorrect flags on clang)
+         */
+        return npy_fmod@c@(a, b);
```

Comment:
What I had missed before is that this also fails the following test for `divmod`: https://github.com/numpy/numpy/blob/6496f1d6abf83cdad37ecdac4453c432ff2f8573/numpy/core/tests/test_umath.py#L579-L580

`fmod(1, 0)` seems to simply not set the correct floating point flags on clang (at least in the version used by CI).  I suppose we could work around clang (or even mark the test as xfail?)